TodoApp/Src/TodoApp.Dependency/Exceptions/DependencyResolutionFailedException.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace TodoApp.Dependency.Exceptions
+{
+    internal class DependencyResolutionFailedException : Exception
```

Comment:
Consider moving the exception to `Contracts` - it is not necessary now, however, you are now unable to catch this very exception and it might be considered a `contract` in literal sense of the word…